ingcsmoreno/tweetbot-r
# Bot Recommendations Improvements

# What does this pull request do?
It improves to the MVP bot by adding the following features:
* recommendations are now queried from a DB
* All interaction with the bot is stored in a DB
* A new hashtag's been added #QuieroSciFi

# How should it be tested?

1. Checkout this branch
2. Get the image created:

    ```
    docker build --build-arg SHA1VER=`git rev-parse HEAD` -t tweet-bot .
    ```
    
3. Run the bot:
  
    ```
    docker run --rm \
    -e CONSUMER_KEY=<CONSUMER_KEY> \
    -e CONSUMER_SECRET=<CONSUMER_SECRET> \
    -e ACCESS_TOKEN=<ACCESS_TOKEN> \
    -e ACCESS_TOKEN_SECRET=<ACCESS_TOKEN_SECRET> \
    tweet-bot:latest
    ```

4. With the bot running, tweet something with hashtag `#quieroscifi` on twitter.
5. Wait a couple of seconds, you should get your tweet replied and retweeted, and you should see something like the following on the logs:

    ```
    2021/02/16 02:54:57 -----------------
    2021/02/16 02:54:57 Tweet ID: 1361509313710333955
    2021/02/16 02:54:57 User: abdelrhmanen1
    2021/02/16 02:54:57 Tweet Text: RT @abdelrhmanen1: *عبر عن حالتك النفسية بصورة* https://t.co/VLMK6sxiPo
    2021/02/16 02:54:57.514688 WARN RESTY Using Basic Auth in HTTP mode is not secure, use HTTPS
    2021/02/16 02:54:57 Tweet Answer: Hay tanto de donde elegir @abdelrhmanen1

    segurísimo que mas de uno te recomendaría Diáspora (1997) de Greg Egan, no vamos a ser la excepción XD
    2021/02/16 02:54:57 Tweet Pic: http://imgur.com/ZGhqdsN.jpg
    ```

# Is related to an Issue?

No no, no issue
![](https://static.wikia.nocookie.net/rolbenigno/images/9/99/Consuela.jpg/revision/latest?cb=20181101221310&path-prefix=es)


# Any extra info?
